### PR TITLE
Make all requests require a JSON body (no more form encoding)

### LINKED ISSUE #10: Form vs JSON encoded request bodies
It would be nice to be able to send all request bodies to the server in JSON format from clients. It is also nice to send certain requests via curl with form encoding. I suggest making a helper that can read either a JSON body, or a form encoded body (based on the request content_type) and return a common dictionary format. For file fields, the JSON format can accept base64 encoded data. We can use this helper for any endpoints that it would be nice to accept both formats.
Current endpoints accepting only form encoded bodies:
- POST /torrent
- POST /auth

Current endpoints accepting only JSON bodies:
- PUT/PATCH /config
- PUT /session/settings

Should all of these accept both? Some of them? Is it a terrible idea?

### PR DESCRIPTION
**Pull request by gazpachoking**
This switches all endpoints to consistently take JSON encoded bodies rather than form encoded. For the POST /torrent endpoint, we'll need to switch the 'file' field to take base 64 encoded data, so that it can be sent via JSON. This would fix #10

TODO:
- [x] Update POST /torrent endpoint
  - [x] Flatten `args` parameters down to base of request
  - [x] Update 'tags' to be passed as 'spritzle.tags'
- [x] Update POST /auth endpoint
- [x] Update docs and examples

Associated spritzle-cli update here: spritzle/spritzle-cli#14

### CODE REVIEW

**Comment by gazpachoking** [comment]
curl examples become somewhat more complicated with this change, not sure how that should go. This works... but isn't so nice.
```
(echo -n '{"file": "'; base64 random_one_file.torrent; echo '", "args": {"save_path": "/tmp"}}') | curl -i -X POST -d @- http://localhost:8080/torrent
```
We could show examples using [httpie](https://httpie.org) which is significantly cleaner, though not as ubiquitous as curl:
```
http POST http://localhost:8080/torrent file="$(base64 random_one_file.torrent)" args:='{"save_path": "/tmp"}'
```

**Review by aresch — APPROVED** [review]
This looks good. Would you mind crafting a PR for spritzle-cli to update it to use this new API?

**Comment by gazpachoking** [comment]
Yeah, I'll make a companion PR for spritzle-cli as well.

**Comment by gazpachoking** [comment]
I think I'm good with the rename. A sub dictionary also sounds fine. Do you expect there to be other spritzle specific stuff coming? Maybe that should wait until there is more than just the one exception? I feel slightly weird about 'file' and 'url' too, which aren't really spritzle specific, but also aren't 1:1 with libtorrent parameters. Don't know that there is anything more clean to be done other than document it though.

**Comment by aresch** [comment]
We use 'spritzle.tags' in the torrent status dictionary so that it's not confused with a libtorrent value. Maybe we should just do that here too.

**Comment by gazpachoking** [comment]
Yeah, that sounds good to me for now. (Although, isn't the torrent status dictionary only for spritzle values? 'spritzle.tags' is the only thing in there right now, no?)

**Comment by gazpachoking** [comment]
A [couple of the examples got really long](https://github.com/gazpachoking/spritzle/blob/0870a9a23fcafadc6fafc03771cbb68ff00dfd13/docs/interface.md#sessionstats-1) to do with the nice way httpie prints the response. Maybe we should add a `...` in the middle to not blow up the scrolling needed for them?

**Comment by gazpachoking** [comment]
I decided truncation was good in those examples and did it.

File: docs/interface.md [hunk 1]
```diff
@@ -28,14 +28,32 @@ Returns a dictionary of the session stats.
 **Example**
 
 ```shell
-$ curl -i -X GET http://localhost:8080/session/stats
+$ http GET http://localhost:8080/session/stats
```
Comment (aresch) [inline docs/interface.md:34]: Could you maybe add something near the top of the document that informs the reader that we're using httpie for examples and a link to the projects website. I'm not certain httpie is as ubiquitous as curl so we should point out where to find it.

Comment (gazpachoking) [reply docs/interface.md:34]: Yeah, good idea, will do.

**Pull request merged**